scripts/multisrc/hotnovelpub/template.ts

```diff
@@ -0,0 +1,276 @@
+    this.lang = metadata.options?.lang || "en";
+  }
+
+  async popularNovels(
+    pageNo: number,
+    { filters, showLatestNovels }: Plugin.PopularNovelsOptions,
+  ): Promise<Plugin.NovelItem[]> {
+    let url = this.apiSite + "/books/";
+    url += showLatestNovels ? "new" : filters?.sort?.value || "hot";
+    if (filters?.category?.value) {
+      url = this.apiSite + "/category/" + filters.category.value;
+    }
+
+    url += "/?page=" + (pageNo - 1) + "&limit=20";
+
+    const result = await fetchApi(url, {
+      headers: {
+        lang: this.lang,
+      },
+    });
+    const json = (await result.json()) as responseNovels;
+    const novels: Plugin.NovelItem[] = [];
+
+    if (json.status && json.data.books.data?.length) {
+      json.data.books.data.forEach((novel) =>
+        novels.push({
+          name: novel.name,
+          cover: this.site + novel.image,
+          path: novel.slug,
+        }),
+      );
+    }
+    return novels;
+  }
+
+  async parseNovel(novelPath: string): Promise<Plugin.SourceNovel> {
+    const result = await fetchApi(this.apiSite + "/book/" + novelPath, {
+      headers: {
+        lang: this.lang,
+      },
+    });
+    const json = (await result.json()) as responseNovel;
+
+    const novel: Plugin.SourceNovel = {
+      name: json.data.book.name,
+      path: novelPath,
+      cover: this.site + json.data.book.image,
+      summary: json.data.book.authorize.description,
+      author: json.data.book.authorize.name,
+      status:
+        json.data.book.status === "updating"
+          ? NovelStatus.Ongoing
+          : NovelStatus.Completed,
+    };
+
+    if (json.data.tags.tags_name?.length) {
+      novel.genres = json.data.tags.tags_name.join(",");
+    }
+
+    if (json.data.chapters?.length) {
+      const chapters: Plugin.ChapterItem[] = [];
+      json.data.chapters.forEach((chapter, chapterIndex) =>
+        chapters.push({
+          name: chapter.title,
+          path: chapter.slug,
+          releaseTime: undefined,
+          chapterNumber: (chapter.index || chapterIndex) + 1,
+        }),
+      );
+
+      novel.chapters = chapters;
+    }
+    return novel;
+  }
+
+  async parseChapter(chapterPath: string): Promise<string> {
+    const body = await fetchApi(this.site + "/" + chapterPath).then((res) => res.text());
+    const loadedCheerio = parseHTML(body);
+
+    let chapterText = loadedCheerio("#content-item").html() || "";
+    if (chapterText) {
+      const result = await fetchApi(
+        this.site + "/server/getContent?slug=" + chapterPath,
+      );
+      const json = (await result.json()) as ChapterType;
+
+      if (json.data) {
+        chapterText += json.data
+          .map((item) => "<p>" + item + "</p>")
+          .join("")
+          .replace(/\n/g, "</p><p>")
+          .replace(/\s/g, " ");
+      }
+    }
+    return chapterText.replace(/\.copy right hot novel pub/g, "");
+  }
+
+  async searchNovels(key_search: string): Promise<Plugin.NovelItem[]> {
```

Comment:
```suggestion
  async searchNovels(key_search: string, pageNo: number): Promise<Plugin.NovelItem[]> {
```
because app implemented page for searching so you have to handle it

Reply:
But the source itself doesn't maintain the pages in search.